src/aiortc/codecs/vpx.py

```diff
@@ -436,6 +436,275 @@ def __update_config(self) -> None:
         self.cfg.rc_target_bitrate = self.__target_bitrate // 1000
         self.__update_config_needed = False
 
+class Vp9Decoder(Decoder):
+    def __init__(self) -> None:
+        self.codec = ffi.new("vpx_codec_ctx_t *")
+        _vpx_assert(
+            lib.vpx_codec_dec_init(self.codec, lib.vpx_codec_vp9_dx(), ffi.NULL, 0)
+        )
+
+        ppcfg = ffi.new("vp8_postproc_cfg_t *")
+        
+        # XXX vp9 uses the same flags
+        ppcfg.post_proc_flag = lib.VP8_DEMACROBLOCK | lib.VP8_DEBLOCK
+        ppcfg.deblocking_level = 3
+        lib.vpx_codec_control_(self.codec, lib.VP8_SET_POSTPROC, ppcfg)
+
+    def __del__(self) -> None:
+        lib.vpx_codec_destroy(self.codec)
+
+    def decode(self, encoded_frame: JitterFrame) -> List[Frame]:
+        frames: List[Frame] = []
+        result = lib.vpx_codec_decode(
+            self.codec,
+            encoded_frame.data,
+            len(encoded_frame.data),
+            ffi.NULL,
+            lib.VPX_DL_REALTIME,
+        )
+        
+        if result == lib.VPX_CODEC_OK:
+            it = ffi.new("vpx_codec_iter_t *")
+            while True:
+                img = lib.vpx_codec_get_frame(self.codec, it)
+                if not img:
+                    break
+                assert img.fmt == lib.VPX_IMG_FMT_I420
+
+                frame = VideoFrame(width=img.d_w, height=img.d_h)
+                frame.pts = encoded_frame.timestamp
+                frame.time_base = VIDEO_TIME_BASE
+
+                for p in range(3):
+                    i_stride = img.stride[p]
+                    i_buf = ffi.buffer(img.planes[p], i_stride * img.d_h)
+                    i_pos = 0
+
+                    o_stride = frame.planes[p].line_size
+                    o_buf = memoryview(cast(bytes, frame.planes[p]))
+                    o_pos = 0
+
+                    div = p and 2 or 1
+                    for r in range(0, img.d_h // div):
+                        o_buf[o_pos : o_pos + o_stride] = i_buf[
+                            i_pos : i_pos + o_stride
+                        ]
+                        i_pos += i_stride
+                        o_pos += o_stride
+
+                frames.append(frame)
+
+        return frames
+
+class Vp9Encoder(Encoder):
+    def __init__(self) -> None:
+        self.cx = lib.vpx_codec_vp9_cx()
```

Comment:
Same q as above: was this function already exposed or did you have to do something for it?

Reply:
This one was already exposed here: https://github.com/geminovc/aiortc/blob/main/src/_cffi_src/build_vpx.py#L213

Reply:
That's convenient!